galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.manager/src/main/java/dev/galasa/cicsts/spi/ITsqProvider.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.cicsts.spi;
+
+import javax.validation.constraints.NotNull;
+
+import dev.galasa.cicsts.ITsqHandler;
+import dev.galasa.cicsts.TsqManagerException;
+import dev.galasa.cicsts.ICicsRegion;
+
+/**
+ * Provides CICS Region related TSQ objects
+ *
+ */
+public interface ITsqProvider {
+    
+    /**
+     * Returns a unique instance of the ICemt per CICS region 
+     * 
+     * @param cicsRegion
+     * @param cicstsManager
+     * @return ITsqHandler object for this CICS region, will a different instance for different regions
+     * @throws TsqManagerException if getTsq() fails
+     */
+    @NotNull
+    ITsqHandler getTsq(ICicsRegion cicsRegion, ICicstsManagerSpi cicstsManager) throws TsqManagerException;
+
```

Comment:
Also, you may know what a TSQ is, wondering if wee should spell it out in the naming of the queue-manipulating object. Like ICicsTemporaryStorageQueue

Reply:
Updated as per the latest discussion to ITsqFactory()